### PR TITLE
Enable BuildKit when building Docker agent image

### PR DESCRIPTION
**Pull request by take-five**
This will help us validate that BuildKit actually works as intended.

### CODE REVIEW

**Comment by take-five** [comment]
It works: https://ci.at.samo.io/job/salemove/job/docker-jenkins-agents/job/PR-13/1/console

**Review by mpi1 — APPROVED** [review]

**Review by kuzaxak — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -37,13 +37,15 @@ def buildAgentImage = { agentName, minorVersion=null ->
...
 withResultReporting(slackChannel: '#tm-inf') {
-  inDockerAgent(containers: [agentContainer(image: 'salemove/jenkins-agent-docker:17.12.0')]) {
+  inDockerAgent(containers: [agentContainer(image: 'salemove/jenkins-agent-docker:19.03.15')]) {
     stage('Checkout code') {
       checkout(scm)
     }
 
     stage("Build jenkins-agent-docker") {
-      buildAgentImage('jenkins-agent-docker')
+      withEnv(['DOCKER_BUILDKIT=1']) {
```
[the remark this answers is not in the document]
Comment (take-five) [reply Jenkinsfile:46]: I think it looks fine here? https://ci.at.samo.io/job/salemove/job/docker-jenkins-agents/job/PR-13/1/console
On my machine when I run it in terminal it looks different (it's interactive, flashy and hides some output, like you said)

[the remark this answers is not in the document]
Comment (mpi1) [reply Jenkinsfile:46]: `--progress=plain` after `build` command should suppress the flashiness.

```
docker-compose build --progress=plain <container_name>

or

docker build --progress=plain .
```

[the remark this answers is not in the document]
Comment (take-five) [reply Jenkinsfile:46]: Yeah, but as far as I understand, BuildKit is smart enough to understand that there's no TTY in Jenkins and prints simple non-flashy reports.

[the remark this answers is not in the document]
Comment (mpi1) [reply Jenkinsfile:46]: > BuildKit is smart enough to understand that there's no TTY in Jenkins

Yes. Just added the commands if you'd locally want to get rid of the disco.

**Pull request merged**